Doesn't compile against PG16

I was prepping this for windows building against PostgreSQL16beta2 and ran into issues compiling against master branch.
PG11 - PG15 compile fine.

I'm suspecting it's this change discussed here - https://www.postgresql.org/message-id/flat/ddcce239-0f29-6e62-4b47-1f8ca742addf%40enterprisedb.com  

It looks like the easiest fix is to include fmgr.h as was done in the patch for many of the contrib extension
https://www.postgresql.org/message-id/attachment/141802/0001-New-header-varatt.h-split-off-from-postgres.h.patch

or a conditional #include "varatt.h" for PG 16 and above.

I'll take a stab at fixing and will provide a pull request if I'm successful.

```
E:/jenkins/h3-pg/h3-pg/h3/src/binding/indexing.c: In function 'h3_cell_to_boundary':
E:/jenkins/h3-pg/h3-pg/h3/src/binding/indexing.c:114:9: warning: implicit declaration of function 'SET_VARSIZE' [-Wimplicit-function-declaration]
  114 |         SET_VARSIZE(polygon, size);
      |         ^~~~~~~~~~~
[26/47] Linking C static library _deps\h3-build\bin\libh3.a
[27/47] Building C object h3/CMakeFiles/postgresql_h3.dir/src/binding/traversal.c.obj
[28/47] Building C object h3/CMakeFiles/postgresql_h3.dir/src/deprecated.c.obj
[29/47] Building C object h3/CMakeFiles/postgresql_h3.dir/src/binding/vertex.c.obj
[30/47] Building C object h3/CMakeFiles/postgresql_h3.dir/src/extension.c.obj
[31/47] Building C object h3_postgis/CMakeFiles/postgresql_h3_postgis.dir/src/init.c.obj
[32/47] Building C object h3/CMakeFiles/postgresql_h3.dir/src/guc.c.obj
[33/47] Building C object h3/CMakeFiles/postgresql_h3.dir/src/opclass_hash.c.obj
[34/47] Building C object h3_postgis/CMakeFiles/postgresql_h3_postgis.dir/src/wkb_bbox3.c.obj
[35/47] Building C object h3/CMakeFiles/postgresql_h3.dir/src/opclass_btree.c.obj
[36/47] Building C object h3/CMakeFiles/postgresql_h3.dir/src/init.c.obj
[37/47] Building C object h3/CMakeFiles/postgresql_h3.dir/src/operators.c.obj
[38/47] Building C object h3/CMakeFiles/postgresql_h3.dir/src/srf.c.obj
[39/47] Building C object h3/CMakeFiles/postgresql_h3.dir/src/type.c.obj
[40/47] Building C object h3_postgis/CMakeFiles/postgresql_h3_postgis.dir/src/wkb_indexing.c.obj
[41/47] Linking C shared module h3\h3.dll
FAILED: h3/h3.dll 
cmd.exe /C "cd . && C:\ming64\mingw64\bin\cc.exe -O3 -DNDEBUG -flto -fno-fat-lto-objects   -shared -o h3\h3.dll -Wl,--major-image-version,0,--minor-image-version,0 include/CMakeFiles/postgresql_h3_shared.dir/error.c.obj h3/CMakeFiles/postgresql_h3.dir/src/binding/edge.c.obj h3/CMakeFiles/postgresql_h3.dir/src/binding/hierarchy.c.obj h3/CMakeFiles/postgresql_h3.dir/src/binding/indexing.c.obj h3/CMakeFiles/postgresql_h3.dir/src/binding/inspection.c.obj h3/CMakeFiles/postgresql_h3.dir/src/binding/miscellaneous.c.obj h3/CMakeFiles/postgresql_h3.dir/src/binding/regions.c.obj h3/CMakeFiles/postgresql_h3.dir/src/binding/traversal.c.obj h3/CMakeFiles/postgresql_h3.dir/src/binding/vertex.c.obj h3/CMakeFiles/postgresql_h3.dir/src/deprecated.c.obj h3/CMakeFiles/postgresql_h3.dir/src/extension.c.obj h3/CMakeFiles/postgresql_h3.dir/src/guc.c.obj h3/CMakeFiles/postgresql_h3.dir/src/init.c.obj h3/CMakeFiles/postgresql_h3.dir/src/opclass_btree.c.obj h3/CMakeFiles/postgresql_h3.dir/src/opclass_hash.c.obj h3/CMakeFiles/postgresql_h3.dir/src/operators.c.obj h3/CMakeFiles/postgresql_h3.dir/src/srf.c.obj h3/CMakeFiles/postgresql_h3.dir/src/type.c.obj  E:/jenkins/postgresql/rel/pg16w64/lib/libpostgres.a  _deps/h3-build/bin/libh3.a  E:/jenkins/postgresql/rel/pg16w64/lib/libpostgres.a  C:/ming64/mingw64/lib/libm.a  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && cd ."
lto-wrapper.exe: warning: using serial compilation of 2 LTRANS jobs
lto-wrapper.exe: note: see the '-flto' option documentation for more information
C:/ming64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\ming64\msys\tmp\cc6aw12f.ltrans0.ltrans.o:<artificial>:(.text+0xf3ca): undefined reference to `SET_VARSIZE'

collect2.exe: error: ld returned 1 exit status
[42/47] Building C object h3_postgis/CMakeFiles/postgresql_h3_postgis.dir/src/wkb_vect3.c.obj
[43/47] Building C object h3_postgis/CMakeFiles/postgresql_h3_postgis.dir/src/wkb_regions.c.obj
[44/47] Building C object h3_postgis/CMakeFiles/postgresql_h3_postgis.dir/src/wkb_linked_geo.c.obj
[45/47] Building C object h3_postgis/CMakeFiles/postgresql_h3_postgis.dir/src/wkb.c.obj
E:/jenkins/h3-pg/h3-pg/h3_postgis/src/wkb.c: In function 'boundary_array_to_wkb':
E:/jenkins/h3-pg/h3-pg/h3_postgis/src/wkb.c:98:9: warning: implicit declaration of function 'SET_VARSIZE' [-Wimplicit-function-declaration]
   98 |         SET_VARSIZE(wkb, VARHDRSZ + size);
      |         ^~~~~~~~~~~
E:/jenkins/h3-pg/h3-pg/h3_postgis/src/wkb.c:100:26: warning: implicit declaration of function 'VARDATA' [-Wimplicit-function-declaration]
  100 |         data = (uint8 *) VARDATA(wkb);
      |                          ^~~~~~~
E:/jenkins/h3-pg/h3-pg/h3_postgis/src/wkb.c:100:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  100 |         data = (uint8 *) VARDATA(wkb);
      |                ^
In file included from E:/jenkins/h3-pg/h3-pg/h3_postgis/src/wkb.c:20:
E:/jenkins/h3-pg/h3-pg/h3_postgis/src/wkb.c:40:32: warning: implicit declaration of function 'VARSIZE'; did you mean 'VERTSIZE'? [-Wimplicit-function-declaration]
   40 |                 (uint8 *)wkb + VARSIZE(wkb) == data, \
      |                                ^~~~~~~
E:/jenkins/h3-pg/h3-pg/include/error.h:23:19: note: in definition of macro 'ASSERT'
   23 |         if (0 == (condition)) ereport(ERROR, ( \
      |                   ^~~~~~~~~
E:/jenkins/h3-pg/h3-pg/h3_postgis/src/wkb.c:103:9: note: in expansion of macro 'ASSERT_WKB_DATA_WRITTEN'
  103 |         ASSERT_WKB_DATA_WRITTEN(wkb, data);
      |         ^~~~~~~~~~~~~~~~~~~~~~~
E:/jenkins/h3-pg/h3-pg/h3_postgis/src/wkb.c: In function 'boundary_to_wkb':
E:/jenkins/h3-pg/h3-pg/h3_postgis/src/wkb.c:117:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  117 |         data = (uint8 *) VARDATA(wkb);
      |                ^
E:/jenkins/h3-pg/h3-pg/h3_postgis/src/wkb.c: In function 'linked_geo_polygon_to_wkb':
E:/jenkins/h3-pg/h3-pg/h3_postgis/src/wkb.c:134:16: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  134 |         data = (uint8 *) VARDATA(wkb);
      |                ^
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,6 +52,8 @@ find_package(PostgreSQL ${POSTGRESQL_VERSION} REQUIRED
 )
 include(AddPostgreSQLExtension)
 
+add_definitions(-DPOSTGRESQL_VERSION_MAJOR=${PostgreSQL_VERSION_MAJOR})
+
 # https://cliutils.gitlab.io/modern-cmake/chapters/testing.html
 if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
   include(CTest)

--- a/h3/src/binding/edge.c
+++ b/h3/src/binding/edge.c
@@ -26,6 +26,10 @@
 #include "type.h"
 #include "srf.h"
 
+#if POSTGRESQL_VERSION_MAJOR >= 16
+#include "varatt.h" //VAR_SIZE and friends moved to here from postgres.h
+#endif
+
 PGDLLEXPORT PG_FUNCTION_INFO_V1(h3_are_neighbor_cells);
 PGDLLEXPORT PG_FUNCTION_INFO_V1(h3_cells_to_directed_edge);
 PGDLLEXPORT PG_FUNCTION_INFO_V1(h3_is_valid_directed_edge);

--- a/h3/src/binding/indexing.c
+++ b/h3/src/binding/indexing.c
@@ -15,6 +15,11 @@
  */
 
 #include <postgres.h>
+
+#if POSTGRESQL_VERSION_MAJOR >= 16
+#include "varatt.h" //VAR_SIZE and friends moved to here from postgres.h
+#endif
+
 #include <h3api.h>
 
 #include <fmgr.h>			 // PG_FUNCTION_INFO_V1

--- a/h3_postgis/src/wkb.h
+++ b/h3_postgis/src/wkb.h
@@ -22,6 +22,10 @@
 
 #include <fmgr.h>
 
+#if POSTGRESQL_VERSION_MAJOR >= 16
+#include "varatt.h" //VAR_SIZE and friends moved to here from postgres.h
+#endif
+
 bytea *
 			boundary_array_to_wkb(const CellBoundary * boundaries, size_t num);
 

